### PR TITLE
Lesson 02 jues jedzej

### PR DESCRIPTION
**Pull request by Resires**
Added lesson 2 before lesson 1 was checked. Should not create any conflicts.

### CODE REVIEW

File: students/uss_tomasz/lesson_01_basics/nowy.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+from math import ceil
```
Comment (jedzej) [inline students/uss_tomasz/lesson_01_basics/nowy.py]: lesson 1 should not be the part of this PR
you probably branched out lesson_02_jues_jedzej from lesson_01_jues_jedzej
you should branch out from master branch instead. You can fix this by rebasing your branch on top of master and pushing the branch again with force

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+else:
+    print("NO")
+
+# Problem Queen move
+x_start = int(input("Starting Column of queen movement"))
+y_start = int(input("Starting Row of queen movement"))
+x_finish = int(input("Finishing Column of queen movement"))
+y_finish = int(input("Finishing Row of queen movement"))
+
+if (abs(x_finish - x_start) == abs(y_finish - y_start)) or (x_start == x_finish or y_start == y_finish):
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py]: just a side note - in python `or` operator has low priority http://www.mathcs.emory.edu/~valerie/courses/fall10/155/resources/op_precedence.html
so some of the brackets above are not needed - this code works in the same way
```
if abs(x_finish - x_start) == abs(y_finish - y_start) or x_start == x_finish or y_start == y_finish:
```

anyway, they can be added for code clarity like in the example here: https://introcs.cs.princeton.edu/python/appendix_precedence/

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+else:
+    print("COMMON")
+
+'''
+https://snakify.org/lessons/for_loop_range/problems/
+'''
+# Problem Factorial
+n = int(input("Number for factorial calculation"))
+factorial = 1
+for i in range(1, n + 1):
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py]: `for i in range(2, n + 1):` does the job as well

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+for i in range(n - 1, 0, -1):
+    factorial_sum += 1
+    factorial_sum *= i
+print("Total sum of following factorials is ", factorial_sum)
+
+# Problem Ladder
+n = int(input("Ladder steps [between 1 and 9] = "))
+for k in range(1, n + 1):
+    for j in range(1, k + 1):
+        print(j, end='')
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py]: find meaningful names for n, k, j variables

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    print()
+
+# Problem Lost card
+N = int(input("Cards quantity = "))
+table = [0 for i in range(N)]
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py]: `0` is a valid int, use `None` if you want to mark missing/invalid/not initialized value
```
> [None]*6
[None, None, None, None, None, None]
```

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+for i in range(1, N):
+    card = int(input("Enter new card (not present in list of already entered cards"))
+    table[card - 1] = 1
+for i in range(N):
+    if table[i] == 0:
+        print("Missing card is ", i + 1)
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py]: this exercise can be solved in single for loop with `list.remove`

**Review by jedzej — CHANGES_REQUESTED** [review]
solid coding, but branching requires fixing

**Comment by Resires** [comment]
Changes applied. 
Performed:
git rebase lesson_02_jues_jedzej master
git push --force

File: students/uss_tomasz/lesson_02_flow_control/main.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+# Problem Ladder
+ladder_height = int(input("Ladder steps [between 1 and 9] = "))
+for row in range(1, ladder_height + 1):
+    for column in range(1, row + 1):
+        print(column, end='')
+    print()
+
+# Problem Lost card
+N = int(input("Cards quantity = "))
+table = [i+1 for i in range(N)]
```
Comment (jedzej) [inline students/uss_tomasz/lesson_02_flow_control/main.py:127]: `table = list(range(1, N+1))`

**Review by jedzej — APPROVED** [review]
Very good naming and clean code after fixes. 

I've fixed the problem with invalid history with the following:
1. go to branch `lesson_02_jues_jedzej`
1. `git rebase master` - after this the branch no longer stems from lesson_01, but from master
1. `git push -f` - to sync remote repository with local changes
```
C:\Users\jedzej\tietopythontraining-basic>git status
On branch lesson_02_jues_jedzej
Your branch is up to date with 'origin/lesson_02_jues_jedzej'.

nothing to commit, working tree clean

C:\Users\jedzej\tietopythontraining-basic>git rebase master
First, rewinding head to replay your work on top of it...
Applying: Added lesson 2 file
Applying: Modify file based on Jedrzej review

C:\Users\jedzej\tietopythontraining-basic>git log
commit 60916951b1a2cd2981c8e2af15d326893b410e68 (HEAD -> lesson_02_jues_jedzej)
Author: Tomek U <1989tomo@gmail.com>
Date:   Sat Sep 28 19:22:18 2019 +0200

    Modify file based on Jedrzej review

commit f05740c2ead0e99e7b509032ecd3da33efb773d5
Author: Tomek U <1989tomo@gmail.com>
Date:   Fri Sep 20 12:30:21 2019 +0200

    Added lesson 2 file

commit bf43d1b83f1ef3fe07a067fd9f57b496397d79b9 (origin/master, origin/HEAD, master)
Merge: 5ae8747 0d17834
Author: Tomasz <1989tomo@gmail.com>
Date:   Sat Sep 28 18:18:36 2019 +0200

    Merge pull request #2 from Resires/lesson_01_jues_jedzej

    Create file, first lesson

commit 0d178349fa5f688c0c8d20bf27b2804588bd19c6
Author: Tomek U <1989tomo@gmail.com>
Date:   Fri Sep 27 15:02:24 2019 +0200

    Modified according to first review by Jedrzej

commit 099f69a565561e3747da365c1a021583b73b9dc7
Author: Tomek U <1989tomo@gmail.com>

C:\Users\jedzej\tietopythontraining-basic>git push -f
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 4 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (12/12), 2.38 KiB | 1.19 MiB/s, done.
Total 12 (delta 6), reused 4 (delta 1)
remote: Resolving deltas: 100% (6/6), completed with 2 local objects.
To https://github.com/Resires/tietopythontraining-basic.git
 + fa56dec...6091695 lesson_02_jues_jedzej -> lesson_02_jues_jedzej (forced update)
```

It's ready to merge.

**Pull request merged**